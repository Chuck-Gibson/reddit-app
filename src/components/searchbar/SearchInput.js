import PropTypes from "prop-types";

export const SearchInput = ({ handleChange, value }) => {
  return (
    <>
      <input
        type="text"
        placeholder="Search something..."
        role="search"
        value={value}
        onChange={handleChange}
      />
    </>
  );
};

SearchInput.propTypes = {
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};
