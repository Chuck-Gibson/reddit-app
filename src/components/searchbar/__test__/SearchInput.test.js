import { SearchInput } from "../SearchInput";
import { render, screen, fireEvent, cleanup } from "@testing-library/react";

afterEach(() => cleanup);

describe("SearchInput component test", () => {
  it("should render", () => {
    const { container } = render(<SearchInput />);
    const inputElement = container.firstChild;
    expect(inputElement).toBeInTheDocument();
  });

  it("it should have a placeholder text", () => {
    const placeholder = "Search something...";
    render(<SearchInput />);
    const inputElement = screen.getByPlaceholderText(/search something.../i);
    expect(inputElement.placeholder).toBe(placeholder);
  });

  it("should render the correct `value`", () => {
    const value = "hello world";
    render(<SearchInput value={value} />);
    const inputElement = screen.getByRole("search");
    expect(inputElement.value).toBe(value);
  });
});
