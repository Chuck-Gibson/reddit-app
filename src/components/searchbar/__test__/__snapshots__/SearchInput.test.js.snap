// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<input
  onChange={[MockFunction]}
  placeholder="Search something..."
  role="search"
  type="text"
  value=""
/>
`;

exports[` 2`] = `
<input
  onChange={[MockFunction]}
  placeholder="Search something..."
  role="search"
  type="text"
  value="foo bar"
/>
`;

exports[`SearchInput snapshot tests renders the proper value of user input 1`] = `
<input
  onChange={[MockFunction]}
  placeholder="Search something..."
  role="search"
  type="text"
  value=""
/>
`;

exports[`SearchInput snapshot tests renders the proper value of user input 2`] = `
<input
  onChange={[MockFunction]}
  placeholder="Search something..."
  role="search"
  type="text"
  value="foo bar"
/>
`;
